test/TestServerProjectsLowLevel/dpg-customization/Customization/DPGClient.cs

```diff
@@ -93,6 +89,50 @@ public virtual Response<Product> PostModel(string mode, Input input, Cancellatio
+        /// <param name="mode"> The mode with which you&apos;ll be handling your returned body. &apos;raw&apos; for just dealing with the raw body, and &apos;model&apos; if you are going to convert the raw body to a customized body before returning to users. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="mode"/> is null. </exception>
+        public virtual async Task<Operation<Product>> LroValueAsync(WaitUntil waitUntil, string mode, CancellationToken cancellationToken = default)
+        {
+            var requestContext = new RequestContext {CancellationToken = cancellationToken};
+            using var scope = ClientDiagnostics.CreateScope("DPGClient.LroValue");
+            scope.Start();
+            try
+            {
+                var lro = await LroAsync(waitUntil, mode, requestContext).ConfigureAwait(false);
+                return LowLevelOperationHelpers.Convert(lro, ClientDiagnostics, "DPGClient.LroValue", r => (Product)r);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <summary> Long running put request that will either return to end users a final payload of a raw body, or a final payload of a model after the SDK has grown up. </summary>
+        /// <param name="waitUntil"> <see cref="WaitUntil.Completed"/> if the method should wait to return until the long-running operation has completed on the service; <see cref="WaitUntil.Started"/> if it should return after starting the operation. For more information on long-running operations, please see <see href="https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/samples/LongRunningOperations.md"> Azure.Core Long-Running Operation samples</see>. </param>
+        /// <param name="mode"> The mode with which you&apos;ll be handling your returned body. &apos;raw&apos; for just dealing with the raw body, and &apos;model&apos; if you are going to convert the raw body to a customized body before returning to users. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="mode"/> is null. </exception>
+        public virtual Operation<Product> LroValue(WaitUntil waitUntil, string mode, CancellationToken cancellationToken = default)
+        {
+            var requestContext = new RequestContext { CancellationToken = cancellationToken };
+            using var scope = ClientDiagnostics.CreateScope("DPGClient.LroValue");
+            scope.Start();
+            try
+            {
+                var lro = Lro(waitUntil, mode, requestContext);
+                return LowLevelOperationHelpers.Convert(lro, ClientDiagnostics, "DPGClient.LroValue", r => (Product)r);
```

Comment:
Nit: could we change the parameter ordering to:

`return LowLevelOperationHelpers.Convert(lro, r => (Product)r, ClientDiagnostics, "DPGClient.LroValue");`

This way the main plot comes first and the less important details are later.